src/Core/Banshee.WebBrowser/Banshee.WebSource/WebView.cs

```diff
@@ -26,64 +26,89 @@
 // THE SOFTWARE.
 
 using System;
-
-using Gtk;
-
-using Hyena;
-using Hyena.Downloader;
-
+using System.Runtime.InteropServices;
 using Banshee.Base;
-using Banshee.IO;
 using Banshee.WebBrowser;
+using Gdk;
+using GLib;
+using Hyena.Downloader;
+using EventArgs = System.EventArgs;
+using EventHandler = System.EventHandler;
+using Log = Hyena.Log;
 
 namespace Banshee.WebSource
 {
     public abstract class WebView : OssiferWebView
     {
-        protected string FixupJavascriptUrl { get; set; }
+        private const float ZoomStep = 0.05f;
         private string fixup_javascript;
         private bool fixup_javascript_fetched;
-
-        public bool IsReady { get; private set; }
-        public bool CanSearch { get; protected set; }
-
-        public event EventHandler Ready;
+        private double smooth_scroll_size;
 
         public WebView ()
         {
             CanSearch = false;
         }
 
-        const float ZOOM_STEP = 0.05f;
+        protected string FixupJavascriptUrl { get; set; }
+
+        public bool IsReady { get; private set; }
+        public bool CanSearch { get; protected set; }
+
+        public event EventHandler Ready;
 
         public void ZoomIn ()
         {
-            Zoom += ZOOM_STEP;
+            Zoom += ZoomStep;
         }
 
         public void ZoomOut ()
         {
-            Zoom -= ZOOM_STEP;
+            Zoom -= ZoomStep;
         }
 
-        protected override bool OnScrollEvent (Gdk.EventScroll scroll)
+        [DllImport ("libgdk-3-0.dll", CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr gdk_event_get_scroll_deltas (IntPtr eventHandle, out double deltaX,
+            out double deltaY);
+
+        protected override bool OnScrollEvent (EventScroll scroll)
         {
-            if ((scroll.State & Gdk.ModifierType.ControlMask) != 0) {
-                Zoom += (scroll.Direction == Gdk.ScrollDirection.Up) ? ZOOM_STEP : -ZOOM_STEP;
+            if ((scroll.State & ModifierType.ControlMask) != 0) {
+                switch (scroll.Direction) {
+                    case ScrollDirection.Up:
+                        ZoomIn ();
+                        break;
+                    case ScrollDirection.Down:
+                        ZoomOut ();
+                        break;
+                    case (ScrollDirection) 4: //GDK_SCROLL_SMOOTH
```

Comment:
little nit-pick, space at the start of line comments please: `// GDK_SCROLL_SMOOTH`